impeller/renderer/backend/vulkan/resource_manager_vk_unittests.cc

```diff
@@ -0,0 +1,23 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <sys/types.h>
+#include <memory>
+#include "flutter/testing/testing.h"
+#include "impeller/renderer/backend/vulkan/resource_manager_vk.h"
+
+namespace impeller {
+namespace testing {
+
+// While expected to be a singleton per context, the class does not enforce it.
+TEST(ResourceManagerVKTest, CreatesANewInstance) {
+  auto const a = ResourceManagerVK::Create();
+  auto const b = ResourceManagerVK::Create();
+  EXPECT_NE(a, b);
+}
+
+// TODO: See what Aaron and Chinmay think can be tested here, and how.
```

Comment:
I think you should have a test that clears out an object.  You can start up the ResourceManagerVK, submit an object that signals a latch in its destructor, then wait for for that latch.